sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -1364,10 +1364,11 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     })
   }
 
-  test("SPARK-6245 JsonRDD.inferSchema on empty RDD") {
+  test("SPARK-6245 JsonRDD.inferSchema on empty Dataset") {
     // This is really a test that it doesn't throw an exception
+    val emptyDataset = spark.createDataset(empty)(Encoders.STRING)
```

Comment:
RDD[_] only has toDS() function added when SQLImplicits applies an implicit conversion to wrap the RDD as a DatasetHolder.
import sparkSession.sqlContext.implicits._